docs/component/oa-verify.md

```diff
@@ -4,8 +4,188 @@ title: Verification SDK (javascript)
+
+### Custom validation
+The `isValid` function will execute over fragments and determine if the fragments produced a valid result. By default the function will return true if a document fulfill the following conditions:
+- The document has NOT been tampered, AND
+- The document has been issued, AND
+- The document as NOT been revoked, AND
+- The issuer identity is valid.
+
+However in some conditions, the result of the function might not be useful: Why is the document not valid ? Is it because it has been tampered ? Or maybe the issuer identity is invalid ?
+
+The function allow to specify as a second parameters the list of types on which to perform the checks. Let's try it with our custom verifiers over the initial wrapped document:
+
+```javascript
+const { verificationBuilder, isValid, openAttestationVerifiers } = require("@govtechsg/oa-verify");
+const document = require("./document.json");
+
+// our custom verifier will be valid only if the document version is not open-attestation/2.0
+const customVerifier = {
+  skip: () => {
+    throw new Error("this verifier is never skipped");
+  },
+  test: () => true,
+  verify: async document => {
+    if (document.version === "open-attestation/2.0") {
+      return {
+        type: "DOCUMENT_INTEGRITY",
+        name: "CustomVerifier",
+        data: document.version,
+        status: "INVALID"
+      };
+    }
+    return {
+      type: "DOCUMENT_INTEGRITY",
+      name: "CustomVerifier",
+      data: document.version,
+      status: "VALID"
+    };
+  }
+};
+
+const verify = verificationBuilder([
+  ...openAttestationVerifiers,
+  customVerifier // see above
+]);
+
+verify(document, { network: "ropsten" }).then(fragments => {
+  console.log(isValid(fragments)); // output false
+  console.log(isValid(fragments, ["DOCUMENT_INTEGRITY"])); // output false
+  console.log(isValid(fragments, ["DOCUMENT_STATUS"])); // output true
+  console.log(isValid(fragments, ["ISSUER_IDENTITY"])); // output true
+});
+```
+
+### Listening to individual verifiers
+The `verify` function has an option to listen to every verifiers individually. It might be useful if you want for instance to provide individual loader on your UI.
+
+```javascript
+const { verify, isValid } = require("@govtechsg/oa-verify");
+const document = require("./document.json");
+
+verify(document, {
+  network: "ropsten",
+  promisesCallback: verifiers => {
+    for (const verifier of verifiers) {
+      verifier.then(fragment =>
+        console.log(
+          `${fragment.name} has been resolved with status ${fragment.status}`
+        )
+      );
+    }
+  }
+}).then(fragments => {
+  console.log(isValid(fragments)); // output true
+});
+```
+
+## Additional information
+- Verification SDK implementation follow our [Verifier ADR](https://github.com/Open-Attestation/adr/blob/master/verifier.md).
+- Found a bug ? Having a question ? Want to share an idea ? Reach us out on the [Github repository](https://github.com/Open-Attestation/oa-verify).`
```

Comment:
Should we also move all of these to the footer of the entire site?

Reply:
you mean github link for every component ?

Reply:
Yea, i noticed that you add this into all the pages. Was wondering if it's possible to add it in all the documentation pages. Seemed like we cant just customise a components (at least not in their docs) to render after each page. So I'm just thinking if we want it big at the footer to remind people they can report bugs or post questions in our github page.

Reply:
yeah why note, I created an issue #13 to tackle this. I think we can continue like this for the moment